### PR TITLE
Update institution program provider facet

### PR DESCRIPTION
**Pull request by devin-mccurdy**
## Description
Change `provider` facet from a hash to an array of hashes.  Previously the hash keys were being used as the program name in vets-website, but that was causing issues due to JSON serialization forcing keys to conform to only valid javascript field names.

This was causing filter checkboxes using those provider names to be invalid search values.

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/19992

## Testing done
Tested locally and QA review

## Screenshots
N/A

## Acceptance criteria
- [x] provider facets data is an array of hashes containing provider name and count 

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

File: spec/controllers/v0/institution_programs_controller_spec.rb [hunk 1]
```diff
@@ -191,8 +191,11 @@
       get(:index)
       facets = JSON.parse(response.body)['meta']['facets']
       expect(facets['provider']).not_to be_nil
-      expect(facets['provider']['acme inc']).not_to be_nil
-      expect(facets['provider']['acme inc']).to eq(1)
+      expect(facets['provider'].size).to eq(5)
+
+      acme_provider = facets['provider'].select { |provider| provider['name'] == 'ACME INC' }.first
+      expect(acme_provider['name']).to eq('ACME INC')
+      expect(acme_provider['count']).to eq(1)
```
Comment (jholton) [inline spec/controllers/v0/institution_programs_controller_spec.rb]: Suggestion:
Change `create(:institution_program, :ca_employer)` to `create_list(:institution_program, 2, :ca_employer)`
Then change these lines to: 
```ruby
acme_providers = facets['provider'].select { |provider| provider['name'] == 'ACME INC' }
expect(acme_providers.count).to eq(1)
expect(acme_providers.first['count']).to eq(2)
```
This shows that the providers are actually being grouped appropriately.
(That is, if there was a bug that caused `facets['provider']` to contain two entries for 'ACME INC', these specs would still pass, which would be bad)

Comment (dmccurdybah) [reply spec/controllers/v0/institution_programs_controller_spec.rb]: Good idea, thanks!

**Review by jholton — APPROVED** [review]
LGTM

**Pull request merged**